CHANGELOG.md

```diff
@@ -1,10 +1,111 @@
 # MultiQC Version History
 
+## [MultiQC v1.25](https://github.com/MultiQC/MultiQC/releases/tag/v1.25) - 2024-09-16
+
+### Highlights
+
+New feature: grouping samples in the General Statistics table.
+
+<img width="400" src="docs/images/changelog/v1.25-grouping-samples.png">
+
+MultiQC does its best to have one row per sample in a table. Some modules, however, may produce multiple results per sample. These will be treated as separate samples alongside the shorter "merged" sample name from downstream steps, resulting in half-empty rows. To address this, we introduce an new configuration option:
+
+```yaml
+table_sample_merge:
+  "R1": "_R1"
+  "R2": "_R2"
+```
+
+It will tell MultiQC that `SAMPLE_R1` and `SAMPLE_R2` should be grouped together into `SAMPLE`. For the modules that have only data for `SAMPLE_R1` and `SAMPLE_R2` - e.g. FastQC - `SAMPLE` will be generated by merging together the data from `SAMPLE_R1` and `SAMPLE_R2`, while individual sample data can be seen by "expanding" a table row.
+
+The way a module merges sample groups for each specific metric can be specified in the module code, e.g.:
+
+```python
+self.general_stats_addcols(
+    ...,
+    group_samples_config=SampleGroupingConfig(
+        cols_to_sum=[ColumnKeyT("total_sequences"), ...],
+        cols_to_weighted_average=[
+            (ColumnKeyT("percent_gc"), ColumnKeyT("total_sequences")),
+            (ColumnKeyT("avg_sequence_length"), ColumnKeyT("total_sequences")),
+            ...
+        ],
+        extra_functions=...,
+    )
+)
+```
+
+For configuration, more sophisticated patterns are supported such as listing multiple strings, or using regular expression with `type: regex`:
+
+```yaml
+table_sample_merge:
+  "Read 1":
+    - "_R1"
+    - type: regex
+      pattern: "[_.-][rR]?1$"
+  "Read 2":
+    - "_R2"
+    - type: regex
+      pattern: "[_.-][rR]?2$"
+```
```

Comment:
This is good, but I think too much detail for the changelog. It's all written in the docs, so better to just point to the relevant sections in the docs.

Also, need to mention here that it's not enabled by default. So:

1. Explain what it is, with the changelog - as you have done ✅ 
2. Name the new config option, say that it's opt-in for now though will be on by default in a future release. Link to the docs about configuration setting for more detail.
3. Explain briefly that modules must independently add support - that we've done one or two and would like to do more, please let us know which modules should do this by creating an issue. Link to the docs about Python functionality and support.

Reply:
Updated!